tasks/managed/run-file-updates/run-file-updates.yaml

```diff
@@ -40,39 +40,129 @@ spec:
+    - name: ociArtifactExpiresAfter
+      description: Expiration date for the trusted artifacts created in the
+        OCI repository. An empty string means the artifacts do not expire.
+      type: string
+      default: "1d"
+    - name: trustedArtifactsDebug
+      description: Flag to enable debug logging in trusted artifacts. Set to a non-empty string to enable.
+      type: string
+      default: ""
+    - name: orasOptions
+      description: oras options to pass to Trusted Artifacts calls
+      type: string
+      default: ""
+    - name: sourceDataArtifact
+      type: string
+      default: ""
+    - name: dataDir
+      description: The location where data will be stored
+      type: string
+      default: $(workspaces.data.path)
     - name: taskGitUrl
       type: string
-      description: The url to the git repo where the release-service-catalog tasks to be used are stored
+      description: The url to the git repo where the release-service-catalog tasks and stepactions to be used are stored
     - name: taskGitRevision
       type: string
       description: The revision in the taskGitUrl repo to be used
   results:
     - name: mergeRequestUrl
       description: URL of MR that was created
+    - description: Produced trusted data artifact
+      name: sourceDataArtifact
+      type: string
   workspaces:
     - name: data
       description: Workspace where the file updates to apply are defined
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+    env:
+      - name: IMAGE_EXPIRES_AFTER
+        value: $(params.ociArtifactExpiresAfter)
+      - name: "ORAS_OPTIONS"
+        value: "$(params.orasOptions)"
+      - name: "DEBUG"
+        value: "$(params.trustedArtifactsDebug)"
   steps:
+    - name: skip-trusted-artifact-operations
+      ref:
+        resolver: "git"
+        params:
+          - name: url
+            value: $(params.taskGitUrl)
+          - name: revision
+            value: $(params.taskGitRevision)
+          - name: pathInRepo
+            value: stepactions/skip-trusted-artifact-operations/skip-trusted-artifact-operations.yaml
+      params:
+        - name: ociStorage
+          value: $(params.ociStorage)
+        - name: workDir
+          value: $(params.dataDir)
+    - name: use-trusted-artifact
+      ref:
+        resolver: "git"
+        params:
+          - name: url
+            value: $(params.taskGitUrl)
+          - name: revision
+            value: $(params.taskGitRevision)
+          - name: pathInRepo
+            value: stepactions/use-trusted-artifact/use-trusted-artifact.yaml
+      params:
+        - name: workDir
+          value: $(params.dataDir)
+        - name: sourceDataArtifact
+          value: $(params.sourceDataArtifact)
     - name: run-script
       image: quay.io/konflux-ci/release-service-utils:0f82be4be43294b6a96846d87ef7f7c0b9e34267
+      onError: continue
```

Comment:
why?

Reply:
since we now don't have a persistent workspace, we want to be able to inspect the workspace after at least 1 of the file-updates has failed. To do that, we need to create the artifacts before we exit with a failure.

If we mark the test as expected to fail, without having the `onError: continue`, we never get to the check-result step to examine.

Reply:
> If we mark the test as expected to fail, without having the onError: continue, we never get to the check-result step to examine.

This isn't a test, it is the task yaml. What are you talking about check-result?

Reply:
From that test
```
              # Fetch all the InternalRequest resources
              allRequests=$(kubectl get InternalRequest -o json)

              # Count the number of InternalRequests
              requestsCount=$(echo "$allRequests" | jq -r '.items | length')

              # Check if the number of InternalRequests is as expected
              if [ "$requestsCount" -ne "1" ]; then
                echo "Unexpected number of InternalRequests. Expected: 1, Found: $requestsCount"
                exit 1
              fi
```
so there aren't even multiple IRs in that test

Reply:
there is clearly a bug in the test...I am re-working it.

Reply:
there was no bug in the tests.

There was an incorrect assumption that we needed to capture the workspace via TA when a test failed. Turns out that with a few changes in `mocks.sh`, we can have our tests checked without anything from the workspace.

With that, I have also removed the `onError: continue` from the **Task**